### PR TITLE
Verify addons installation and deletion as part of tkg clusterclass tests

### LINKED ISSUE #2456: TC07: AWS Cluster - Able to create/delete CC Guest cluster on AWS and verify if all addons are successfully deployed
Complete two parts:
- [x] Create/delete ClusterClass guest cluster on AWS
- [ ] Verify the following addons packages are successfully deployed:
   - CNI: antrea.tanzu.vmware.com
   - Kapp: kapp-controller.tanzu.vmware.com
   - AddittionalPackages: metrics-server.tanzu.vmware.com
   - AddittionalPackages: secretgen-controller.tanzu.vmware.com
   - AddittionalPackages: pinniped.tanzu.vmware.com

Addons are listed at https://github.com/vmware-tanzu/tanzu-framework/blob/main/apis/run/v1alpha3/clusterbootstraptemplate_types.go#L21-L29. They are,

- CNI
- CSI
- CPI
- Kapp
- AdditionalPackages

Since CSI and CPI are only used in vSphere, I will not include them for these AWS integration tests.

As for AdditionalPackages, I think it would be sufficient to test that these additional packages are successfully deployed:

```
additionalPackages:
  - refName: metrics-server.tanzu.vmware.com.0.5.1+vmware.1-tkg.2-zshippable
  - refName: secretgen-controller.tanzu.vmware.com.0.8.0+vmware.1-tkg.1-zshippable
  - refName: pinniped.tanzu.vmware.com.0.12.1+vmware.1-tkg.1-zshippable
 ```

Use the test https://github.com/vmware-tanzu/tanzu-framework/blob/package-based-lcm/pkg/v1/tkg/test/tkgctl/aws_cc/aws_cc_suite_test.go as an example to go off of.

Use the library created from #2388 to iterate through addons and verify they were installed correctly.

### PR DESCRIPTION
**Pull request by shyaamsn**
### What this PR does / why we need it
Verify addons installation and deletion as part of tkg clusterclass tests

Adds the following:
- [x] Verify if packages are reconciled successfully on management cluster and workload cluster
- [x] Verify if clusterbootstrap is created correctly from clusterbootstraptemplate for both management and workload cluster
- [x] Verify if packages installation status is reflected correctly in clusterbootstrap.Status for both management and workload cluster
- [x] Verify if addon finalizer and machine termination hooks are set correctly for workload cluster
- [x] Verify if cluster resources are deleted correctly after cluster is deleted

### Which issue(s) this PR fixes

Fixes #2456 #2487 

### Describe testing done for PR
AWS Clusterclass pipeline in github passes

### Release note

```release-note
package-based-lcm: Verify addons installation and deletion as part of tkg clusterclass tests
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by shivaani0505 — APPROVED** [review]
LGTM, thanks for the PR.

File: pkg/v1/tkg/test/tkgctl/shared/package.go [hunk 1]
```diff
@@ -80,19 +158,134 @@ func checkClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName
...
+
+// checkClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
+func checkClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
+	log.Infof("Verify addons on workload cluster %s with management cluster %s", wcClusterName, mcClusterName)
+
+	var clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap
+	if isManagementCluster {
+		clusterBootstrap = getClusterBootstrap(ctx, mccl, mcClusterNamespace, mcClusterName)
+	} else {
+		clusterBootstrap = getClusterBootstrap(ctx, mccl, wcClusterNamespace, wcClusterName)
```
Comment (ggpaue) [inline pkg/v1/tkg/test/tkgctl/shared/package.go:177]: do we have to get clusterbootstrap from workload cluster?

Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/shared/package.go:177]: No, true that the resource exists on the management cluster itself, but it is under each cluster's name/namespace anyway

**Comment by maralavi** [comment]
Thanks for the PR

**Review by maralavi — APPROVED** [review]

**Pull request merged**